### ISSUE TITLE
Offline: Don't fetch Soil ID when offline

### ISSUE THREAD

**Issue by DerekCaelin** [techmatters/terraso-mobile-client#2415]
Currently, the site popover and the soil ID tile displays "Error" when  viewed and the app is in Offline. The app should store soil ID information and present the most recent version of truth if it is not able to fetch new information.

#### Current & Expected temporary location Popover
<img width="150px" src="https://github.com/user-attachments/assets/d1a24731-3f57-4c10-9a61-3943a3c55174">

<img width="150px" alt="image" src="https://github.com/user-attachments/assets/71d3746e-ca72-43e9-9524-cbaa59d0cdd0">

#### Current & Expected Offline Soil ID Tile 
<img width="150px" src="https://github.com/user-attachments/assets/ea7b40f4-67a2-4cda-9f6e-476ef95767e2">

<img width="150px" alt="image" src="https://github.com/user-attachments/assets/87219b46-a264-4433-ab97-f01ac3ad2530">

### Acceptance Criteria
- [ ] If offline,
   - [x] Site popover shows last determined soil ID and Ecological Site ID
   - [x] Soil ID tile shows last determined soil ID and Ecological Site ID
   - [x] Soil ID page shows information last fetched from the soil ID algorithm

**Comment by DerekCaelin** [comment]
I think this was assigned to me for a cleaning up... I should have asked an engineer to review. @tm-ruxandra it sounds like you are already doing work to satisfy this user story, is that correct?

**Comment by tm-ruxandra** [comment]
@DerekCaelin correct, I was handling this as part of #2372

**Comment by CourtneyLee333** [comment]
What I notice is that if the temp site popover remains open when I go offline:
- Soil ID Top Match and ES Top Match remain in the popover
- Elevation displays "Not Available Offline"

But if I close the temp popover and go offline (or go offline first and then close the popover), enter the exact same coordinates:
- All 3 fields on the popover display "Not Available Offline"

Soil ID works as expected for sites.

**Comment by tm-ruxandra** [comment]
@CourtneyLee333 I can't reproduce that behavior -- for `37.43,-122` I'm able to repeatedly exit the popover and re-enter coordinates to see previously-loaded soil ID data. I'm curious which coordinates you're using to test?

**Comment by DerekCaelin** [comment]
@tm-ruxandra I can reproduce courtney's experience.

1. Be online, search for coordinates `38,-122`
2. In temporary popover, see top match for soil ID, ESD, elevation
3. Go offline
4. See top match for Soil ID and ESD, see elevation "Not available offline"

and then

5. Close window. Search again for  `38,-122`
6. See top match for Soil ID and ESD, see elevation "Not available offline"

**Comment by tm-ruxandra** [comment]
@DerekCaelin @CourtneyLee333 I think there's some miscommunication here: Courtney said that both the Soil ID match entries and elevation are showing as "not available offline", but Derek said that only elevation is showing as "not available offline". I am seeing what Derek is seeing: elevation is not available offline, but soil ID matches for coordinates persist if I enter the same coordinates.

**Comment by DerekCaelin** [comment]
You're right, my experience reloading the site is different from Courtney's.

**Comment by CourtneyLee333** [comment]
Okay, it seems to me that the thing causing our different experience is that I'm tapping on the map to start with, as opposed to entering coordinates in the search bar. It sounds like you both are entering coordinates to start with, which allows the soil info to persist for me as well. 

Can you reproduce the error if you tap the map to get your temp location in the first step? Should we expect soil info to persist if you do this?

**Comment by tm-ruxandra** [comment]
@CourtneyLee333 Okay, I think I understand now. It sounds like there's an issue with coordinate precision (since tapping on the map might load a very precise location, but our UI only renders coordinates to 5 decimal places.) I do think that your workflow should be supported, so I can submit an update to this to make the caching of location-based matches consistent with how the UI presents coordinates to the user.
